gtime/causality/granger_causality.py

```diff
@@ -0,0 +1,225 @@
+    
+    # Parameters of the fitted linear regression model
+    linreg_params = list(linreg_joint.coef_)
+    linreg_params.append(linreg_joint.intercept_)
+    linreg_params = np.array([linreg_params])
+    constraint_params = np.dot(constraint_matrix, linreg_params.T)
+    params_diff = constraint_params - value_restriction
+        
+    pseudoinv_data = pseudoinv_extended(np.append(data_joint.values, np.ones((len(data_joint.values), 1)), axis=1))
+
+    # Covariance matrix
+    scale = linreg_joint_residues / dof_joint
+    covar = np.dot(pseudoinv_data, np.transpose(pseudoinv_data)) * scale
+    covar = np.dot(constraint_matrix, np.dot(covar, constraint_matrix.T))
+    invcov = np.linalg.pinv(covar)
+
+    f = (np.dot(np.dot(params_diff.T, invcov), params_diff) / len(constraint_matrix))[0, 0]
+    pvalue = stats.f.sf(f, len(constraint_matrix), dof_joint)
+    
+    result_df = pd.DataFrame()
+    result_df['F-test'] = [f, pvalue, int(dof_joint), int(max_shift)]
+    result_df.index = ['F-value', 'p-value', 'degrees of freedom', 'number of shifts']
+    return result_df
+
+STAT_TESTS = {'ssr_f': _ssr_f,
+              'ssr_chi2': _ssr_chi2,
+              'likelihood_chi2': _likelihood_chi2,
+              'zero_f': _zero_f}
+
+class GrangerCausality(BaseEstimator):
+    """Class to check for Granger causality between two time series, i.e. 
+    to check if a time series X causes Y: X->Y.
+
+    Parameters
+    ----------
+    target_col : str
+        The column to use as the reference, i.e. the time series Y
+
+    x_col : str
+        The column to test for Granger causality, i.e. the time 
+        series X.
+
+    max_shift : int
+        The maximal number of shifts to check for Granger causality. 
+
+    statistics : list, optional, default: ['ssr_f']
+        The statistical test(s) to perform for Granger causality. A list with elements
+        from the set: 'ssr_f' (sum squared residuals with F-test), 'ssr_chi2' (sum squared 
+        residuals with chi square test), 'likelihood_chi2' (likelihood ratio test with 
+        chi square distribution), 'zero_F' (F-test that all lag coefficients of 
+        the time series X are zero).
+
+    Attributes
+    ----------
+    results_ : list
+        A list of pandas dataframes with the results for all the listed tests in 'statistics'.
+
+    Examples
+    --------
+    >>> from gtime.causality.granger_causality import GrangerCausality
+    >>> import pandas.util.testing as testing
+    >>> data = testing.makeTimeDataFrame(freq="s", nper=1000)
+    >>> gc = GrangerCausality(target_col='A', x_col='B', max_shift=10, statistics=['ssr_f']).fit(data)
+    >>> gc.results_[0]
+                        ssr F-test
+    F-value               0.372640
+    p-value               0.958527
+    degrees of freedom  969.000000
+    number of shifts     10.000000
+
+    """
+
+    def __init__(self, 
+                 target_col: str, 
+                 x_col: str, 
+                 max_shift: int, 
```

Comment:
If we want to be coherent with the other causality tests, we should put a default value for the `max_shift`, like 10.

Reply:
Ok, I set it to 10.